### PR TITLE
Switch to Array.Empty and use Marshal.Copy instead of manual copying

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: src/Foundation/NSUuid.cs [hunk 1]
```diff
@@ -40,11 +40,12 @@ unsafe public NSUuid (byte [] bytes) : base (NSObjectFlag.Empty)
 		public byte [] GetBytes ()
 		{
 			byte [] ret = new byte [16];
-			
-			IntPtr buf = Marshal.AllocHGlobal (16);
-			GetUuidBytes (buf);
-			Marshal.Copy (buf, ret, 0, 16);
-			Marshal.FreeHGlobal (buf);
+
+			unsafe {
```
Comment (chamons) [inline src/Foundation/NSUuid.cs]: This one doesn't seem to match the others in description, though it looks reasonable to me.

Comment (Therzok) [reply src/Foundation/NSUuid.cs]: I think this was supposed to be in a different PR. I can split it out if you want.

File: src/Foundation/NSUuid.cs [hunk 2]
```diff
@@ -40,11 +40,12 @@ unsafe public NSUuid (byte [] bytes) : base (NSObjectFlag.Empty)
...
+				fixed (byte* buf = ret) {
+					GetUuidBytes ((IntPtr)buf);
```
Comment (rolfbjarne) [inline src/Foundation/NSUuid.cs]:
```suggestion
					GetUuidBytes ((IntPtr) buf);
```

**Comment by mandel-macaque** [comment]
Known issue: https://github.com/xamarin/maccore/issues/581

**Pull request merged**